xv7-kernel/src/memory.rs

```diff
@@ -79,9 +81,336 @@ impl<'map> FrameDeallocator<Size4KiB> for BitmapFrameAllocator<'map> {
     }
 }
 
+#[cfg(not(feature = "frame-allocator-bitmap"))]
 lazy_static! {
     pub static ref FRAME_ALLOCATOR: Mutex<BitmapFrameAllocator<'static>> = {
         static mut MAP: [u8; MAX_FRAMES_SUPPORTED / 8] = [0; MAX_FRAMES_SUPPORTED / 8];
         Mutex::new(BitmapFrameAllocator::new(unsafe { &mut MAP }))
     };
 }
+
+/// Buddy system allocator
+///
+/// another frame allocator implemented with buddy system
+pub struct BuddyFrameAllocator<'zone> {
+    zone: &'zone mut BuddyZone,
+    frames: &'zone mut [BuddyFrame; MAX_FRAMES_SUPPORTED],
+}
+
```

Comment:
Can we separate different allocator implementations into different sub mods? like,

```rust
pub mod buddy;
pub mod bitmap;
```